pdc/apps/auth/views.py

```diff
@@ -152,12 +150,8 @@ def get_users_and_groups(resource_permission):
         pass
 
     # get all users
-    users_list = []
-    users = models.User.objects.all()
-    for user in users:
-        uer_group_list = [group.name for group in user.groups.all()]
-        if list(set(groups_list).intersection(set(uer_group_list))):
-            users_list.append(str(user.username))
+    users_set = {str(user.username) for user in get_user_model().objects.filter(groups__name__in=groups_list)}
```

Comment:
Need to test it to make sure it works correctly. I didn't test it. Is this what you need in logic?